### PR TITLE
Propagate `check` return type to `wait()`

### PR DESCRIPTION
**Pull request by happz**
Given that wait() returns what `check` callback returned, we can
propagate callbacks return value - wrapped by `Result` - to return
value of `wait()`, it must be the same type.

### CODE REVIEW

**Comment by happz** [comment]
Fun with types begins :)

It's quite simple: `wait()` either ends with an exception or returns what `check` returned, therefore return value type of `wait()` must match that of `check`. This gives us yet another rule that can be checked in our code, catching possible mismatch between `check` and expected outcome of `wait()`.

**Review by marmolak — APPROVED** [review]

**Review by akhait — APPROVED** [review]
LGTM

File: gluetool/utils.py [hunk 1]
```diff
@@ -1504,11 +1508,19 @@ def match(self, s, multiple=False):
         raise GlueError("Could not match string '{}' with any pattern".format(s), sentry_fingerprint=[s])
 
 
-WaitCheckType = Callable[[], Result[Any, Any]]
+# The type of `wait` is a bit complicated: callback returns `Result` instance with either valid value A, or
+# error B. `wait` us supposed to return `A` when finished successfully - it raises an timeout exception otherwise
```
Comment (optak) [inline gluetool/utils.py:1512]: `wait` IS supposed .. ?

**Review by optak — COMMENTED** [review]
LGTM

**Review by optak — APPROVED** [review]

**Review by thrix — APPROVED** [review]
LGTM

**Pull request merged**